src/main/java/storage/ExcludeInSerializationAnnotationExclusionStrategy.java

```diff
@@ -0,0 +1,17 @@
+package storage;
+
+import com.google.gson.ExclusionStrategy;
+import com.google.gson.FieldAttributes;
+
+/**
+ * Exclusion strategy that excludes class attributes with the {@code @ExcludeInSerialization} annotation
+ */
+class ExcludeInSerializationAnnotationExclusionStrategy implements ExclusionStrategy {
+    public boolean shouldSkipClass(Class<?> clazz) {
+        return clazz.getAnnotation(ExcludeInSerializationAnnotation.class) != null;
```

Comment:
For my own understanding: skips class aka does not serialize if the the class has the annotation.